magma/debug.py

```diff
@@ -44,3 +59,28 @@ def unwire(i, o=None, debug_info=None, keep_wired_when_contexts=False):
             debug_info = get_debug_info(3)
         return fn(i, o, debug_info, keep_wired_when_contexts)
     return unwire
+
+
+def _helper_function(fn):
+
+    @functools.wraps(fn)
+    def _wrapper(*args, **kwargs):
+        extra_frames_to_skip_stack = _get_extra_frames_to_skip_stack()
+        # We keep track of the previous frames_to_skip value since the stack is
+        # additive.
+        try:
+            offset = extra_frames_to_skip_stack.peek()
+        except IndexError:
+            offset = 0
+        # NOTE(rsetaluri): We need to skip an extra frame (1 + 1 = 2) since we
+        # are returning a wrapped function (decorated).
+        extra_frames_to_skip_stack.push(2 + offset)
+        # TODO(rsetaluri): Catch potential exceptions.
+        ret = fn(*args, **kwargs)
```

Comment:
@leonardt what's the right way to catch exceptions here?

```python
try:
    ret = fn(*args, **kwargs)
except Exception as e:
    extra_frames_to_skip_stack.pop()
    raise e from None
```
?

Seems like there's some better pattern to do this?